### PR TITLE
🌱 Add loggercheck linter and fix findings

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ linters:
   - govet
   - importas
   - ineffassign
+  - loggercheck
```
Comment (sbueringer) [inline .golangci.yml:39]: Validates for complete k/v pairs (i.e. we get a finding if we add an odd number of arguments)

**Review by killianmuldoon — COMMENTED** [review]
This looks really useful.
/lgtm

**Review by killianmuldoon — COMMENTED** [review]
/area ci

**Comment by sbueringer** [comment]
Yup especially (IIRC) because the logger panics when these cases are hit

File: test/infrastructure/docker/internal/docker/manager.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ func createNode(ctx context.Context, opts *nodeCreateOpts) (*types.Node, error)
 		}
 	}
 
-	log.V(6).Info("Container run options: %+v", runOptions)
+	log.V(6).Info(fmt.Sprintf("Container run options: %+v", runOptions))
```
Comment (chrischdi) [inline test/infrastructure/docker/internal/docker/manager.go:175]:
```suggestion
	log.V(6).Infof("Container run options: %+v", runOptions)
```

Comment (sbueringer) [reply test/infrastructure/docker/internal/docker/manager.go:175]: logr.Logger does not have an Infof func

Comment (chrischdi) [reply test/infrastructure/docker/internal/docker/manager.go:175]: ah, got it 👍

Comment (chrischdi) [reply test/infrastructure/docker/internal/docker/manager.go:175]: (feel free to resolve)

**Pull request merged**